pulpcore/pulpcore/app/models/content.py

```diff
@@ -138,6 +138,15 @@ def natural_key(self):
         """
         return tuple(getattr(self, f) for f in self.natural_key_fields())
 
+    def natural_key_dict(self):
```

Comment:
This should be simplified to:
```
return {f: getattr(self, f) for f in self.natural_key_fields()}
```
This is simple enough for the caller to just do.  I don't think this (1-line) method adds value and should not be added.

Reply:
The use case I think about with this is plugin writers wanting to query for a unit they have in memory using `filter()` or `Q()`. There are two places I needed to do this for in DeclarativeVersion. I wanted to state it's purpose and see what you thought.

Reply:
That is the use case I'd imaged as well so my opinion hasn't changed.